zebra-network/src/peer_set/limit.rs

```diff
@@ -178,8 +178,6 @@ impl Drop for ConnectionTracker {
 
         // We ignore disconnected errors, because the receiver can be dropped
         // before some connections are dropped.
-        //
-        // TODO: This channel will be bounded by the connection limit (#1850, #1851, #2902).
         let _ = self.close_notification_tx.send(ConnectionClosed);
```

Comment:
This security constraint was implemented:

```suggestion
        // # Security
        //
        // This channel is actually bounded by the inbound and outbound connection limit.
        let _ = self.close_notification_tx.send(ConnectionClosed);
```